lib/models/device-type.ts

```diff
@@ -16,8 +16,78 @@ limitations under the License.
 
 import type { InjectedDependenciesParam, PineOptions } from '..';
 import { DeviceType } from '../types/models';
+import { Partials, Contract } from '../types/contract';
 import { mergePineOptions } from '../util';
 import * as errors from 'balena-errors';
+import * as Handlebars from 'handlebars';
+import cloneDeep = require('lodash/cloneDeep');
+
+// REPLACE ONCE HOST OS CONTRACTS ARE GENERATED THROUGH YOCTO
+import { BalenaOS } from './balenaos-contract';
+
+const traversingCompile = (
+	partials: Partials,
+	initial: Contract,
+	keys: string[],
+): Contract => {
+	return Object.keys(partials).reduce(
+		(interpolated: Contract, partialKey) => {
+			const current = partials[partialKey];
+			if (Array.isArray(current)) {
+				let location: any = interpolated;
+				for (const key of keys) {
+					location = location[key];
+				}
+				// if array of partials, compile the template
+				location[partialKey] = current
+					.map((partial: string) => Handlebars.compile(partial)(interpolated))
```

Comment:
I would expect TS to be able to properly infer this
```suggestion
					.map((partial) => Handlebars.compile(partial)(interpolated))
```